### PR TITLE
Add support for custom ubuntu series when selecting runners

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Since we are planning to support custom image for self-hosted runners, we should remove the hard-coded `jammy` label. This PR proposes to replace it with a new input argument `self-hosted-runner-image`

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -29,6 +29,10 @@ on:
         type: string
         description: Label for selecting the self-hosted runners.
         default: "large"
+      self-hosted-runner-series:
```
Comment (cbartz) [inline .github/workflows/test.yaml:32]:
```suggestion
      self-hosted-runner-image:
```
We are moving into custom image support for the self-hosted runners, so there may be different types of images supported. We also use the term `image` more than series in this context.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -29,6 +29,10 @@ on:
...
+        type: string
+        description: Flavor of the requested runner. Supports only 'jammy' or 'noble'.
```
Comment (cbartz) [inline .github/workflows/test.yaml:34]:
```suggestion
        description: Image of the requested runner. Supports only 'jammy' or 'noble'.
```

File: .github/workflows/test.yaml (hunk 1 above)
Comment (Thanhphan1147) [reply .github/workflows/test.yaml:32]: I see, updating the input name, thanks!

**Comment by Thanhphan1147** [comment]
I think we can roll-out this change first on only the `test` workflow, and port it to other workflows once it has stabilized.

**Review by cbartz — DISMISSED** [review]
LGTM . Can you add the complexity label to the PR and provide a one-liner in the PR description? Thank you very much.

**Review by arturo-seijas — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]

**Review by weiiwang01 — APPROVED** [review]

**Review by gregory-schiano — APPROVED** [review]

**Pull request merged**